src/lib/nvme_pt_ioctl.c

```diff
@@ -361,13 +367,62 @@ void opal_deinit_pt(struct sed_device *dev)
 	}
 
 	if (dev->priv != NULL) {
+		struct opal_device *opal_dev = dev->priv;
+
+		if (opal_dev->req_buf != NULL) {
+			free(opal_dev->req_buf);
+			opal_dev->req_buf = NULL;
+		}
+
 		free(dev->priv);
 		dev->priv = NULL;
 	}
 
 	opal_parser_deinit();
 }
 
+static uint64_t tper_prop_to_val(struct sed_device *dev, const char *tper_prop_name)
+{
+	struct sed_tper_properties *tper = &dev->discv.sed_tper_props;
+
+	for (int j = 0; j < NUM_TPER_PROPS; j++) {
+		if (strncmp(tper_prop_name, tper->property[j].key_name,
+				strlen(tper_prop_name)) == 0)
+			return tper->property[j].value;
+	}
+
+	SEDCLI_DEBUG_MSG("Invalid TPer property name!\n");
+
+	return -EINVAL;
+}
+
+static int resize_io_buf(struct opal_device *dev, uint64_t size)
+{
+	uint8_t *ptr;
+
+	if (dev == NULL)
+		return -EINVAL;
+
+	if (dev->req_buf != NULL)
+		free(dev->req_buf);
+
+	/*
+	 * Allocate memory for request and response buffers in a
+	 * single malloc request and split it later.
+	 */
+	ptr = malloc(sizeof(*dev->req_buf) * 2 * size);
+	if (!ptr)
+		return -ENOMEM;
+	memset(ptr, 0, sizeof(*ptr));
```

Comment:
I think you intended to zero whole newly allocated buffer:
memset(ptr, 0, sizeof(*ptr) * 2 * size)

Reply:
My bad. will fix it